src/Dashboard/Prompty/PlayPrompty/PromptyConsole/PromptyConsole.js

```diff
@@ -0,0 +1,508 @@
+import { useState, useEffect } from "react";
+import { getFirestore, doc, setDoc, getDoc } from "firebase/firestore";
+import { useDocument } from "react-firebase-hooks/firestore";
+import Loader from "../../../../Generic/Loader/Loader";
+import { MySwal } from "../../../../Generic/Notify";
+import { useNavigate } from "react-router-dom";
+
+async function textCompletionApiCall(prompt) {
+  const functions = getFunctions();
+  const callOpenAi = httpsCallable(functions, "generateAiResponse");
+  return await callOpenAi(prompt);
+}
+
+async function moderationApiCall(prompt) {
+  const functions = getFunctions();
+  const callOpenAi = httpsCallable(functions, "moderatePrompt");
+  return await callOpenAi(prompt);
+}
+
+const PromptyConsole = (props) => {
+  const [roleText, setRoleText] = useState("");
+  const [contextText, setContextText] = useState("");
+  const [taskText, setTaskText] = useState("");
+  const [openPromptText, setOpenPromptText] = useState("");
+  const [loader, setLoader] = useState(false);
+  const [promptScaffoldMode, setPromptScaffoldMode] = useState(true);
+  const [promptyInstanceData, setPromptyInstanceData] = useState();
+  const [responsesData, setResponsesData] = useState([]);
+  const [allowGenerate, setAllowGenerate] = useState(false);
+  const [userRole, setUserRole] = useState();
+
+  const navigate = useNavigate();
+
+  const [instanceData, loading, error] = useDocument(
+    doc(
+      getFirestore(),
+      "prompty",
+      props.instanceCode,
+      "instances",
+      props.identifier
+    ),
+    {
+      snapshotListenOptions: { includeMetadataChanges: true },
+    }
+  );
+
+  useEffect(() => {
+    instanceData !== undefined && instanceDataFunctions();
+    function instanceDataFunctions() {
+      setPromptyInstanceData(instanceData.data());
+    }
+  }, [instanceData]);
+
+  useEffect(() => {
+    promptyInstanceData?.generations !== undefined &&
+      setResponsesData(promptyInstanceData.generations);
+  }, [promptyInstanceData]);
+
+  useEffect(() => {
+    if (promptScaffoldMode) {
+      if (
+        roleText !== "" &&
+        contextText !== "" &&
+        taskText !== "" &&
+        loader !== true
+      ) {
+        setAllowGenerate(false);
+      } else {
+        setAllowGenerate(true);
+      }
+    } else {
+      if (openPromptText !== "" && loader !== true) {
+        setAllowGenerate(false);
+      } else {
+        setAllowGenerate(true);
+      }
+    }
+  }, [
+    roleText,
+    contextText,
+    taskText,
+    loader,
+    openPromptText,
+    promptScaffoldMode,
+  ]);
+
+  useEffect(() => {
+    async function getRole() {
+      const db = getFirestore();
+      const docRef = doc(db, "users", props.identifier);
+      const docSnap = await getDoc(docRef);
+      const data = docSnap.data();
+      if (data !== undefined) {
+        setUserRole(data.role);
+      }
+    }
+    getRole();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Why did you disable eslint here? You can remove the second parameter if you don't want to specify the dependencies